nautobot/core/testing/filters.py

```diff
@@ -155,6 +165,116 @@ def test_tags_filter(self):
+                rel_obj = getattr(obj, filter_field_name).first()
+            else:
+                rel_obj = getattr(obj, filter_field_name)
+
+            if "__" not in nested_field:
+                self._assert_non_nested_q_filter(rel_obj, model_field_name, nested_field, lookup_method)
+            else:
+                self._assert_nested_q_filter(rel_obj, model_field_name, nested_field, lookup_method)
+
+        def _assert_non_nested_q_filter(
+            self,
+            obj,
+            model_field_name,
+            filter_field_name,
+            lookup_method,
+        ):
+            """
+            Asserts the validity of a non-nested filter in the queryset.
+
+            Args:
+                obj: The object on which the filter is applied.
+                model_field_name: The name of the field in the model.
+                filter_field_name: The name of the field being filtered.
+                lookup_method: The method used for the lookup.
+            """
+            self._assert_valid_filter_predicates(obj, filter_field_name)
+
+            # Create random 5 char string to append to attribute, used for icontains partial lookup
+            lookup = "".join(random.choices(string.ascii_lowercase, k=5))  # noqa: S311 # pseudo-random generator
+            obj_field_value = getattr(obj, filter_field_name)
+            updated_attr = obj_field_value + lookup
+            setattr(obj, filter_field_name, updated_attr)
+            obj.save()
+            # if lookup_method is iexact use the full updated attr
+            if lookup_method == "iexact":
+                lookup = updated_attr
+                model_queryset = self.queryset.filter(**{f"{model_field_name}": lookup})
+            else:
+                model_queryset = self.queryset.filter(**{f"{model_field_name}__icontains": lookup})
+            params = {"q": lookup}
+            filterset_result = self.filterset(params, self.queryset)
+            self.assertTrue(filterset_result.is_valid())
+            self.assertQuerysetEqualAndNotEmpty(filterset_result.qs, model_queryset)
+
+        def _get_relevant_filterset_queryset(self, queryset, *filter_params):
+            """Gets the relevant queryset based on filter parameters."""
+
+            q_query = Q()
+            for param in filter_params:
+                q_query &= Q(**{f"{param}__isnull": False})
+            queryset = queryset.filter(q_query)
+
+            if not queryset.count():
+                raise ValueError(
+                    f"Cannot find valid test data for {queryset.model.__name__} with params {filter_params}"
+                )
+            return queryset
+
+        def test_q_filter_valid(self):
+            """Test the `q` filter based on attributes in `filter_predicates`."""
+
+            if not isinstance(self.filterset.declared_filters.get("q"), SearchFilter):
+                self.skipTest("`q` filter is not a SearchFilter")
```

Comment:
IMO this should be a failure of the test. Though I recognize that may cause apps to start failing upstream testing.

Reply:
I don't believe `SearchFilter` would be very usefull in some `FilterSets`, such as `IPAddressFIlterset` and `PrefixFIlterset`.

Reply:
Fair point, I hadn't thought of that. Is there a subset of `Filter` types that we *could* assert that includes those cases?

Reply:
There are currently test cases for each of these `q` fields in their respective Test classes.